pkgs/development/web/playwright/firefox.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, fetchzip
+, wrapGAppsHook
+, autoPatchelfHook
+, patchelfUnstable
+, gtk3
+, gnome
+, alsa-lib
+, dbus-glib
+, xorg
+, curl
+, libva
+, pciutils
+, suffix
+, revision
+}:
+let
+  suffix' = if suffix == "linux"
+            then "ubuntu-22.04"
+            else suffix;
+in
+stdenv.mkDerivation {
+  name = "firefox";
```

Comment:
aint it possible to reuse pkgs/applications/networking/browsers/firefox-bin/default.nix ? like overriding it

Reply:
Inheriting most of the attributes now.

Reply:
Please squash the review commits into the initial ones. That makes reviewing a lot easier

Reply:
I am bit annoyed that we make hydra build yet another firefox. Is this just patching a prebuilt executable ?

Reply:
Are you concern due to disk space usage? We are not building from src, we are downloading a pre-built binary and running patchelf on it. 

Would be nice to experiment with upstream browsers here, but as playwright is quite sensitive w.r.t. to browser versions (and at least historically they sometimes patched browsers iirc, don't know if they still do) i believe their binaries are a good default